subsys/usb/usb_c/usbc_tc_common.c

```diff
@@ -29,6 +29,14 @@ void tc_subsys_init(const struct device *dev)
 	smf_set_initial(SMF_CTX(tc), &tc_states[TC_DISABLED_STATE]);
 }
 
+/**
+ * @brief Directly trigger an error recovery
+ */
+void tc_start_error_recovery(const struct device *dev)
+{
+	tc_set_state(dev, TC_ERROR_RECOVERY_STATE);
```

Comment:
I can't remember what our naming convention is for internal vs API functions for the individual state machines, but is it safe to change the TC state asynchronously from the PE?

Reply:
I agree that this could be a problem and I'v changed the code.

Reply:
I can't remember exactly how this code was before, but it looks like `pe_drs_send_swap_run` is still going to indirectly call `tc_set_state` while the TC is not running. Downstream, I think we'd typically have `tc_start_error_recovery` set a flag that the TC checked on the next iteration.

Reply:
I messed up my rebasing and merged the change in the following commit. I'll fix it and upload another PR